examples/near_global_ocean_simulation.jl

```diff
@@ -0,0 +1,299 @@
+using Oceananigans: architecture, on_architecture
+using ClimaOcean
+using ClimaOcean.ECCO
+using ClimaOcean.OceanSimulations
+using ClimaOcean.OceanSeaIceModels
+using CairoMakie
+using OrthogonalSphericalShellGrids
+
+using CFTime
+using Dates
+
+# ### Grid Configuration 
+#
+# We define a global grid with a horizontal resolution of 1/4 degree and 40 vertical levels. 
+# The grid is a `LatitudeLongitudeGrid` capped at 75°S to 75°N.
+# We use an exponential vertical spacing to better resolve the upper ocean layers. The total depth of the domain is set to 6000 meters.
+# Finally, we specify the architecture for the simulation, which in this case is a GPU.
+
+arch = GPU() 
+
+z_faces = exponential_z_faces(Nz=40, depth=6000)
+
+Nx = 1440
+Ny = 600
+Nz = length(z_faces) - 1
+
+grid = LatitudeLongitudeGrid(arch;
+                             size = (Nx, Ny, Nz),
+                             halo = (7, 7, 7),
+                             z = z_faces, 
+                             latitude  = (-75, 75),
+                             longitude = (0, 360))
+
+# ### Bathymetry and Immersed Boundary
+#
+# We retrieve the bathymetry from the ETOPO1 data, ensuring a minimum depth of 10 meters
+# (depths shallower than this are considered land). The `interpolation_passes` parameter
+# specifies the number of passes to interpolate the bathymetry data. A larger number
+# results in a smoother bathymetry. We also remove all connected regions (such as inland
+# lakes) from the bathymetry data by specifying `connected_regions_allowed = 2` (Mediterrean
+# sea an North sea in addition to the ocean).
+
+bottom_height = retrieve_bathymetry(grid; 
+                                    minimum_depth = 10,
+                                    interpolation_passes = 5,
+                                    connected_regions_allowed = 2)
+ 
+grid = ImmersedBoundaryGrid(grid, GridFittedBottom(bottom_height)) 
+
+bathymetry = deepcopy(Array(interior(bottom_height, :, :, 1)))
+bathymetry[bathymetry .>= 0] .= NaN
+
+fig = Figure(size = (800, 400))
+ax = Axis(fig[1, 1], title = "Bathymetry [m]")
+hm = heatmap!(axis, bathymetry, colormap = :deep, colorrange = (-6000, 0))
+cb = Colorbar(fig[1, 2], hm)
+hidedecorations!(ax)
+
+save("bathymetry.png", fig)
+nothing #hide
+
+# ![](bathymetry.png)
+
+# ### Ocean Model Configuration
+#
+# To configure the ocean simulation, we use the `ocean_simulation` function from ClimaOcean.jl. This function allows us to build
+# an ocean simulation with default parameters and numerics. The defaults include:
+# - CATKE turbulence closure for vertical mixing, see [`CATKEVerticalDiffusivity`](@ref)
+# - WENO-based advection scheme for momentum in the vector invariant form, see [`WENOVectorInvariant`](@ref)
+# - WENO-based advection scheme for tracers, see [`WENO`](@ref)
+# - `SplitExplicitFreeSurfaceSolver` with 75 substeps, see [`SplitExplicitFreeSurface`](@ref)
+# - TEOS-10 equation of state, see [`TEOS10EquationOfState`](@ref)
```

Comment:
the docs refs here (i.e. `(@ref)`) won't work because they mean to point to the docs of Oceananigans which is a different package.